src/v/kafka/server/group.cc

```diff
@@ -61,11 +62,13 @@ group::group(
   , _num_members_joining(0)
   , _new_member_added(false)
   , _conf(conf)
-  , _partition(std::move(partition))
+  , _p(p)
+  , _partition(p != nullptr ? p->partition : nullptr)
```

Comment:
from what I can tell `p` is always non-null. what was the case that it wasn't?

Reply:
unit tests, one of them passes `null`